hotsos/core/checks.py

```diff
@@ -766,3 +769,46 @@ def core(self):
         # go fetch
         self.all
         return self._core_snaps
+
+
+class SSLCertificatesChecksBase(object):
+    """
+    This class handles all the ssl certificate checks we want to do.
+    """
+
+    def __init__(self, certificate_path):
+        """
+        @param certificate_path: path to the certificate that we want to load
+        in order to gather information from.
+        If the file in the path fails to read, it raises an exception.
+        """
+        self.certificate_path = certificate_path
+        try:
+            with open(self.certificate_path, "rb") as fd:
+                self.certificate = fd.read()
+        except IOError:
+            log.debug("Unable to read SSL certificate file %s",
+                      self.certificate_path)
+            raise
+
+    @property
+    def certificate_expire_date(self):
```

Comment:
I think it might make sense to have a SSLCertificate class with these properties that way you can have more than one cert and have this class focus on checks rather than providing cert properties.